app/controllers/calendar_controller.rb

```diff
@@ -1,11 +1,22 @@
 class CalendarController < ApplicationController
 
   def index
-    @days = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri']
+    week_start = params[:date] ? Date.parse(params[:date]).at_beginning_of_week : Date.today.beginning_of_week
+    week_end = week_start + CalendarHelper::WEEK_LENGTH - 1
+    @days = (week_start..week_end).to_a
+
     start_time = Time.now.at_beginning_of_day
     end_time = Time.now.at_end_of_day
     step = 30.minutes
-    @times = (start_time.to_i..end_time.to_i).step(step).collect { |time| Time.at time }
+    @times = time_interval(start_time, end_time, step)
+
+    @events = Event.where("start_time >= ? AND end_time <= ?", week_start, week_end)
```

Comment:
As talked with @lkudyba - let's move it to class & add specs #skinnycontrollers.